hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/Text.java

```diff
@@ -302,7 +300,10 @@ public void clear() {
    */
   private boolean ensureCapacity(final int capacity) {
     if (bytes.length < capacity) {
-      bytes = new byte[capacity];
+      // Try to expand  the backing array by the factor of 1.5x  
+      // (by taking the current size + diving it by half)
```

Comment:
"Try to expand the backing array by a factor of 1.5x
(by taking the current size + dividing it by half)"
Fix the warning too.